test/src/userobjects/ExampleShapeElementUserObject.C

```diff
@@ -0,0 +1,103 @@
+/****************************************************************/
+/*               DO NOT MODIFY THIS HEADER                      */
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*           (c) 2010 Battelle Energy Alliance, LLC             */
+/*                   ALL RIGHTS RESERVED                        */
+/*                                                              */
+/*          Prepared by Battelle Energy Alliance, LLC           */
+/*            Under Contract No. DE-AC07-05ID14517              */
+/*            With the U. S. Department of Energy               */
+/*                                                              */
+/*            See COPYRIGHT for full restrictions               */
+/****************************************************************/
+
+#include "ExampleShapeElementUserObject.h"
+#include "libmesh/quadrature.h"
+
+template<>
+InputParameters validParams<ExampleShapeElementUserObject>()
+{
+  InputParameters params = validParams<ShapeElementUserObject>();
+  params.addRequiredCoupledVar("u", "first coupled variable");
+  params.addRequiredCoupledVar("v", "second coupled variable");
+  return params;
+}
+
+ExampleShapeElementUserObject::ExampleShapeElementUserObject(const InputParameters & parameters) :
+    ShapeElementUserObject(parameters),
+    _u_value(coupledValue("u")),
+    _u_var(coupled("u")),
+    _v_value(coupledValue("v")),
+    _v_var(coupled("v"))
+{
+}
+
+void
+ExampleShapeElementUserObject::initialize()
+{
+  _integral = 0.0;
+
+  // Jacobian term storage is up to the user. One option is using an std::vector
+  // We resize it to the total number of DOFs in the system and zero it out.
+  // WARNING: this can be large number (smart sparse storage could be a future improvement)
+  // if (_currently_computing_jacobian)
```

Comment:
Oh I commented this because the way it was called was not right. I can turn this on. We will have to turn on the check in `finalize` & `threadJoin` as well.